types/src/transaction/authenticator.rs

```diff
@@ -107,7 +108,7 @@ impl TransactionAuthenticator {
             Self::Ed25519 {
                 public_key,
                 signature,
-            } => signature.verify(raw_txn, public_key),
+            } => signature.verify(raw_txn, public_key.validate()?),
```

Comment:
This will indeed make a difference, so a key is only validated when we verify a signature. With the current model of `TransactionAuthenticator` this indeed makes sense, because we verify the signature at the very end and only once. If in the future tx verification get smarter (i.e., batch  transactions of the same `signer` together to reuse the validated key multiple times) we might need to improvise again.
All in all, as we don't necessarily expect many transactions of the same user in a block (LedgerInfo) - (well we'll see), I agree with the design decision here.

Reply:
> If in the future tx verification get smarter (i.e., batch transactions of the same signer together to reuse the validated key multiple times or cache previously seen public keys) we might need to improvise again.

This PR should handle those cases already, if you notice the `Validatable` type caches the validated type `T` (in this case the validated `Ed25519PublicKey`) such that subsequent calls to `Validatable::validate` should be free (assuming the key is valid and if its invalid we'd bail out earlier anyway I suppose)